packages/contracts/contracts/foundry/SphinxUtils.sol

```diff
@@ -886,65 +887,108 @@ contract SphinxUtils is SphinxConstants, StdUtils {
         }
     }
 
-    /**
-     * @notice Converts an array of `AccountAccess` structs to an array of transactions
-     *         that can be executed from a Gnosis Safe.
-     */
-    function makeGnosisSafeTransactions(
-        address _gnosisSafeAddress,
-        Vm.AccountAccess[] memory _accountAccesses
-    ) external pure returns (GnosisSafeTransaction[] memory) {
-        // First, we'll get the number of transactions that will be sent from the Gnosis Safe. We'll
-        // need this value because Solidity requires us to define memory arrays with an explicit
-        // number of elements.
-        uint256 numTxns = 0;
-        for (uint256 i = 0; i < _accountAccesses.length; i++) {
-            Vm.AccountAccess memory accountAccess = _accountAccesses[i];
-
-            // We'll add the `AccountAccess` struct if its sender (i.e. `accessor`) is the Gnosis
-            // Safe, and it's either a `Call` or `Create` kind.
-            bool isActionInput = accountAccess.accessor == _gnosisSafeAddress &&
-                (accountAccess.kind == VmSafe.AccountAccessKind.Call ||
-                    accountAccess.kind == VmSafe.AccountAccessKind.Create);
-
-            if (isActionInput) {
-                numTxns += 1;
+    function getNumRootAccountAccesses(
+        Vm.AccountAccess[] memory _accesses,
+        address _safeAddress
+    ) private pure returns (uint256) {
+        uint256 count = 0;
+        for (uint256 i = 0; i < _accesses.length; i++) {
+            Vm.AccountAccess memory access = _accesses[i];
+
+            if (isRootAccountAccess(access, _safeAddress)) {
+                count += 1;
             }
         }
+        return count;
+    }
+
+    function isRootAccountAccess(
+        Vm.AccountAccess memory _access,
+        address _safeAddress
+    ) private pure returns (bool) {
+        return
+            _access.accessor == _safeAddress &&
+            (_access.kind == VmSafe.AccountAccessKind.Call ||
+                _access.kind == VmSafe.AccountAccessKind.Create);
+    }
+
+    function getNumNestedAccountAccesses(
+        Vm.AccountAccess[] memory _accesses,
+        uint256 _rootIdx,
+        address _safeAddress
+    ) private pure returns (uint256) {
+        uint256 count = 0;
+        for (uint256 i = _rootIdx + 1; i < _accesses.length; i++) {
+            Vm.AccountAccess memory access = _accesses[i];
+            if (!isRootAccountAccess(access, _safeAddress)) {
+                count += 1;
+            }
+        }
+        return count;
+    }
+
+    function parseAccountAccesses(
```

Comment:
This is the main function that parses the account accesses. I added some tests in `SphinxUtils.t.sol` for edge cases. We already have the necessary coverage for normal deployments in `deploy.spec.ts`